### PR TITLE
Fix/s2pro sglang voice clone logits alignment

### PR DESCRIPTION
**Pull request by minleminzui**
## Motivation

[fix] improve S2-Pro voice cloning consistency on SGLang

## Modifications
- preserve `references` in `/v1/audio/speech` so voice-cloning conditioning is not dropped
- add E2E mock-pipeline test to lock the references pass-through behavior
- align S2-Pro semantic-token logits numerics to bf16 grid and use compile mode without cudagraphs for better logits consistency

```
# launch the server
CUDA_VISIBLE_DEVICES=0 python -m sglang_omni.cli.cli serve \
  --model-path /data/ckpt/s2-pro \
  --config examples/configs/s2pro_tts.yaml \
  --host 0.0.0.0 \
  --port 8023
  
  
# request
curl -X POST http://127.0.0.1:8023/v1/audio/speech \
-H "Content-Type: application/json" \
-d '{
  "input": "你好",
  "references": [{"audio_path": "nan1.wav", "text": "xxx"}]
}' \
--output output.wav
```
## Related Issues

#152

## Accuracy Test

a4180690919cd662492cdd5638f551a1

## Checklist

- [ ] Format your code according with pre-commit.
- [ ] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

### CODE REVIEW

**Comment by wmj9346464543** [comment]
@minleminzui 
How to make a streaming call? Adding the stream parameter during testing does not achieve streaming return
